testbed/core/views.py

```diff
@@ -72,3 +82,200 @@ def index(request):
         'destination_actor': user_actors.filter(role=Actor.ROLE_DESTINATION).first(),
         'oauth_form': oauth_form,
     })
+
+
+# OAuth Testing Views
+
+def oauth_callback(request):
+    """
+    Handle the callback from the OAuth server
+    This displays the authorization code or error message
+    """
+    # Get query parameters
+    code = request.GET.get('code')
+    error = request.GET.get('error')
+    error_description = request.GET.get('error_description')
+    state = request.GET.get('state')
+    
+    # Initialize context with received parameters
+    context = {
+        'code': code,
+        'error': error,
+        'error_description': error_description,
+        'state': state
+    }
+    
+    # Validate the state parameter to prevent CSRF attacks
+    if not state:
+        logger.warning("OAuth callback received with no state parameter")
+        context['error'] = 'invalid_state'
+        context['error_description'] = 'No state parameter provided'
+    else:
+        is_valid_state = validate_state_from_session(request, state)
+        if not is_valid_state:
+            logger.warning("OAuth callback received with invalid state parameter")
+            context['error'] = 'invalid_state'
+            context['error_description'] = 'Invalid state parameter'
+    
+    # Log successful authorization
+    if code and not error and not context.get('error'):
+        logger.info("Successfully received authorization code in callback")
+    
+    return render(request, 'oauth_callback.html', context)
+
+@login_required
+def test_authorization_view(request):
+    """
+    Test view to simulate initiating an OAuth authorization flow.
+    This will redirect to the authorization endpoint with appropriate parameters.
+    """
+    # Get the user's OAuth application (this is normally used by the destination service)
+    application = get_user_application(request.user)
+    
+    # Use our own callback URL for the test flow
+    scheme = request.scheme
+    host = request.get_host()
+    redirect_uri = f"{scheme}://{host}/callback"
```

Comment:
I understand the convenience of this here for initially getting things to work, but lets move the template redirect_URI and updating the application redirect_URIs somewhere else.  One view is that we don't want to update the Application object in the middle of a view call - the Application object should default to this template URI when we create it instead, for example. But another view is that even if we want to update it during this view, it's still the job of the model - the Application can return (or set) its own redirect_uris.